google-cloud-storage/acceptance/storage/bucket_uniform_bucket_level_access_test.rb

```diff
@@ -198,4 +199,22 @@
     file_default_acl.reload!
     _(file_default_acl.acl.readers).must_equal ["allUsers"]
   end
+
+  it "sets public_access_prevention to enforced and is unable to modify bucket ACL rules" do
+    refute bucket.public_access_prevention_enforced?
+    bucket.public_access_prevention = :enforced
+    assert bucket.public_access_prevention_enforced?
+    err = expect do
+      bucket.acl.public!
+    end.must_raise Google::Cloud::FailedPreconditionError
```

Comment:
@shaffeeullah This integration test is currently failing using the google-cloud-ruby CI project:

```
  1) Failure:
Google::Cloud::Storage::Bucket::uniform_bucket_level_access::storage#test_0013_sets public_access_prevention to enforced and is unable to modify bucket ACL rules [/Users/quartzmo/code/google/codez/google-cloud-ruby/google-cloud-storage/acceptance/storage/bucket_uniform_bucket_level_access_test.rb:207]:
Google::Cloud::FailedPreconditionError expected but nothing was raised.
```

Reply:
@quartzmo is this test still failing?

Reply:
Yes, I still get this failure using `helical-zone-771`:

```
  1) Failure:
Google::Cloud::Storage::Bucket::uniform_bucket_level_access::storage#test_0013_sets public_access_prevention to enforced and is unable to modify bucket ACL rules [/Users/quartzmo/code/google/codez/google-cloud-ruby/google-cloud-storage/acceptance/storage/bucket_uniform_bucket_level_access_test.rb:207]:
Google::Cloud::FailedPreconditionError expected but nothing was raised.
```

Reply:
This acceptance test is now passing with the service correctly raising `FailedPreconditionError`, when run with both the google-cloud-ruby CI project and my dev project.